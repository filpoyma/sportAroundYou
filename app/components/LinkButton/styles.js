import { css } from 'styled-components';

import { COLORS, OTHER } from '@/utils/styleConstants';

const LinkButtonStyles = css`
  display: inline-block;
  padding: 0;
  background: none;
  color: ${COLORS.TEXT.HOVER};
  font: 14px/16px var(--ff);
  border: none;
  cursor: pointer;
  ${OTHER.HOVER_DARKEN};
`;

export default LinkButtonStyles;
