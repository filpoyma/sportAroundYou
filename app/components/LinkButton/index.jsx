import PropTypes from 'prop-types';
import React, { memo } from 'react';

import injectStyles from '@/utils/injectStyles';

import styles from './styles';

const LinkButton = ({ children, ...rest }) => (
  <button type="button" {...rest}>
    {children}
  </button>
);

LinkButton.propTypes = {
  children: PropTypes.node.isRequired
};

LinkButton.defaultProps = {};

LinkButton.displayName = 'LinkButtonComponent';

export default injectStyles(
  memo(LinkButton, (prevProps, newProps) => prevProps.children === newProps.children),
  styles
);
