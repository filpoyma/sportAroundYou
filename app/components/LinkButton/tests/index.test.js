import { render } from '@testing-library/react';
import React from 'react';

import LinkButton from '../index';

describe('<LinkButton />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<LinkButton />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(<LinkButton />);

    expect(firstChild).toMatchSnapshot();
  });
});
