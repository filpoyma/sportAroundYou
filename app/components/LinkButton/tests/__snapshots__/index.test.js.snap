// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> Should render and match the snapshot 1`] = `
<button
  class="sc-1cbh0i-0 GNNOL"
  type="button"
/>
`;
